src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/RemoveAzureSynapseSqlPoolRestorePoint.cs

```diff
@@ -0,0 +1,114 @@
+        [ResourceNameCompleter(ResourceTypes.Workspace, nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string WorkspaceName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DeleteByNameParameterSet, HelpMessage = HelpMessages.SqlPoolName)]
+        [ResourceNameCompleter(ResourceTypes.SqlPool, nameof(WorkspaceName))]
+        [ValidateNotNullOrEmpty]
+        public string SqlPoolName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DeleteByParentObjectParameterSet, HelpMessage = HelpMessages.SqlPoolRestorePointName)]
+        [Parameter(Mandatory = true, ParameterSetName = DeleteByNameParameterSet, HelpMessage = HelpMessages.SqlPoolRestorePointName)]
+        [ResourceNameCompleter(
+            ResourceTypes.SqlPoolRestorePoint,
+            nameof(ResourceGroupName),
+            nameof(WorkspaceName),
+            nameof(SqlPoolName))]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ValueFromPipeline = true, ParameterSetName = DeleteByParentObjectParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.SqlPoolObject)]
+        [ValidateNotNull]
+        public PSSynapseSqlPool SqlPoolObject { get; set; }
+
+        [Parameter(ValueFromPipeline = true, ParameterSetName = DeleteByInputObjectParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.SqlPoolRestorePointObject)]
+        [ValidateNotNull]
+        public PSRestorePoint InputObject { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = DeleteByResourceIdParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.SqlPoolRestorePointResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.PassThru)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.AsJob)]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.SqlPoolObject))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.SqlPoolObject.Id);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.WorkspaceName = resourceIdentifier.ParentResource;
+                this.WorkspaceName = this.WorkspaceName.Substring(this.WorkspaceName.LastIndexOf('/') + 1);
+                this.SqlPoolName = resourceIdentifier.ResourceName;
+            } 
+
+            if (this.IsParameterBound(c => c.InputObject))
+            {
+                this.Name = this.InputObject.Id.Substring(this.InputObject.Id.LastIndexOf('/') + 1);
+                this.ResourceId = this.InputObject.Id.Substring(0, this.InputObject.Id.Substring(0, this.InputObject.Id.LastIndexOf("/")).LastIndexOf("/"));
```

Comment:
>this.ResourceId [](start = 16, length = 15)

I see you set this.ResourceId here. Will line 87-95 be reached in this case?

Reply:
it will reach